scientific/spectroscopy/specpr.ksy

```diff
@@ -0,0 +1,254 @@
+                Geometric albedo   = 2000000001
+      - id: sphase
+        type: s4
+        doc: >
+          The phase angle between iangl and eangl
+          (Integer*4 number, in arc-seconds*1500).
+          (180 degrees=972000000; -180 deg <= phase <= 180 deg)
+                integrating sphere = 2000000000
+      - id: iwtrns
+        type: s4
+        doc: "Weighted number of runs (the number of runs of the spectrum with the minimum runs which was used in processing this spectrum, integer*4)."
+      - id: itimch
+        type: s4
+        doc: "The time observed in the sample beam for each half chop in milliseconds (for chopping spectrometers only). (integer*4)"
+      - id: xnrm
+        type: f4
+        doc: "The band normalization factor. For data scaled to 1.0, multiply by this number to recover photometric level (32 bit real number)."
+      - id: scatim
+        type: f4
+        doc: "The time it takes to make one scan of the entire spectrum in seconds (32 bit real number)."
+      - id: timint
+        type: f4
+        doc: "Total integration time (usually=scatime * nruns) (32 bit real number)."
+      - id: tempd
+        type: f4
+        doc: "Temperature in degrees Kelvin (32 bit real number)."
+      - id: data
+        type: f4
+        doc: "The spectral data (256 channels of 32 bit real data numbers)."
+        repeat: expr
+        repeat-expr: 256
+    instances:
+      phase_angle_arcsec:
+        value: sphase / 1500.
+        doc: "The phase angle between iangl and eangl in seconds"
+  data_continuation:
+    seq:
+      - id: cdata
+        type: f4
+        repeat: expr
+        repeat-expr: 383
+        doc: "The continuation of the data values (383 channels of 32 bit real numbers)."
+  text_initial:
+    seq:
+      - id: ids
+        type: identifiers
+      - id: itxtpt
+        type: u4
+        doc: "Text data record pointer. This pointer points  to a data record where additional text may be may be found."
+      - id: itxtch
+        type: s4
+        doc: "The number of text characters (maximum= 19860)."
+      - id: itext
+        type: str
+        size: 1476
+        doc: "1476 characters of text.  Text has embedded newlines so the number of lines available is limited only by the number of characters available."
+  text_continuation:
+    seq:
+      - id: tdata
+        type: str
+        size: 1532
+        doc: "1532 characters of text."
+  record:
+    seq:
+    - id: icflag
+      type: icflag
+      doc: "Total number of bytes comprising the document."
+    - id: content
+      size: 1536 - 4
+      type:
+          switch-on: icflag.type
+          cases:
+            'record_type::data_initial': data_initial
+            'record_type::data_continuation': data_continuation
+            'record_type::text_initial': text_initial
+            'record_type::text_continuation': text_continuation
+enums:
+  record_type: # if I use 0b notation it doesn't work

```

Comment:
We don't support expression language for enum IDs, so it's processed by YAML, and YAML [only supports decimal, hex and octal](http://yaml.org/spec/1.2/spec.html#id2761509) integer literals. In theory, we can add conversion from strings like `0b1010111` manually...